### ISSUE TITLE
Make "Whole Word" and "Regex" modes mutually exclusive

### ISSUE THREAD

**Issue by lee-dohm** [atom/find-and-replace#1044]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

See https://github.com/atom/find-and-replace/issues/1042 and https://github.com/atom/find-and-replace/pull/976.

### Steps to Reproduce

1. Launch Atom
1. Open the find-and-replace panel with <kbd>Cmd+F</kbd>
1. Click the "Regex" mode button to turn it on
1. Click the "Whole Word" mode button to turn it on

**Expected behavior:** "Regex" mode turns off

**Actual behavior:** "Regex" mode remains enabled

### Resolution

Make it so that one can only have "Regex" mode on or "Whole Word" mode on, but not both.

**Comment by jedwards1211** [comment]
There are two PRs open to fix this by wrapping the regex in a non-capturing group (one was mine because I failed to find the first PR); are you sure it's necessary to make the two mutually exclusive?

**Comment by tribut** [comment]
Given that two people arrived at the exact same changeset to resolve this problem properly I frankly find the explanation given in https://github.com/atom/find-and-replace/pull/976 rather puzzling. Please reconsider merging one of the PRs. If necessary I could also convert my examples from the PR to a regression test. The functionality is actually helpful if implemented correctly and the fact that we found the bug means we tried to use it.

**Comment by Evpok** [comment]
A better solution imo would be to make the regex mode ignore (and grey out the checkbox of) whole-word and ignore-case as those don't make much sense in regex mode anyway

**Comment by tribut** [comment]
I disagree, @Evpok. Ignore-case is a very common modifier to regexes (can't think of a single editor that provides regex search without it) and as for whole-word: See above.
